Mathlib/Util/CompileInductive.lean

```diff
@@ -0,0 +1,179 @@
+/-
+Copyright (c) 2023 Parth Shastri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Parth Shastri
+-/
+import Lean.Compiler.CSimpAttr
+import Lean.Elab.Predefinition
+
+/-!
+# Define the `#compile inductive` command.
```

Comment:
Can you explain what this command does at a high level? (not necessarily including implementation details)